perl/t/lib/Util.pm

```diff
@@ -49,28 +49,45 @@ sub contents {
 sub db_setup {
     $Wallet::Config::DB_DDL_DIRECTORY = 'sql/';
     if (-f 't/data/test.database') {
-        open (DB, '<', 't/data/test.database')
-            or die "cannot open t/data/test.database: $!";
-        my $driver = <DB>;
-        my $info = <DB>;
-        my $user = <DB>;
-        my $password = <DB>;
-        chomp ($driver, $info);
-        chomp $user if $user;
-        chomp $password if $password;
-        $Wallet::Config::DB_DRIVER = $driver;
-        $Wallet::Config::DB_INFO = $info;
-        $Wallet::Config::DB_USER = $user if $user;
-        $Wallet::Config::DB_PASSWORD = $password if $password;
+        db_setup_from_test_database();
     } else {
+        db_setup_sqlite();
+    }
+}
 
-        # If we have a new SQLite db by default, disable version checking.
-        $ENV{DBIC_NO_VERSION_CHECK} = 1;
+sub db_setup_from_test_database {
+    open (DB, '<', 't/data/test.database')
+        or die "cannot open t/data/test.database: $!";
+    my $driver = <DB>;
+    my $info = <DB>;
+    my $user = <DB>;
+    my $password = <DB>;
+    chomp ($driver, $info);
+    chomp $user if $user;
+    chomp $password if $password;
+    $Wallet::Config::DB_DRIVER = $driver;
+    $Wallet::Config::DB_INFO = $info;
+    $Wallet::Config::DB_USER = $user if $user;
+    $Wallet::Config::DB_PASSWORD = $password if $password;
+}
 
-        $Wallet::Config::DB_DRIVER = 'SQLite';
-        $Wallet::Config::DB_INFO = 'wallet-db';
-        unlink 'wallet-db';
-    }
+sub db_setup_sqlite {
+    # If we have a new SQLite db by default, disable version checking.
+    $ENV{DBIC_NO_VERSION_CHECK} = 1;
+
+    $Wallet::Config::DB_DRIVER = 'SQLite';
+    $Wallet::Config::DB_INFO = 'wallet-db';
+    unlink 'wallet-db';
+}
+
+
+sub setup_initialize {
+    my $admin;
+    eval {
+        local $ENV{DBIC_NO_VERSION_CHECK} = 1;
```

Comment:
I don't understand this change or why it's in an `eval`. Isn't this handled by `db_setup_sqlite`? And swallowing all exceptions seems wrong.